kubectl-rook-ceph.sh

```diff
@@ -226,14 +227,220 @@ function path_cm_rook_ceph_operator_config() {
+  set +eu
+  info_msg "Extracting the monmap"
+  KUBECTL_NS_CLUSTER exec deploy/rook-ceph-mon-$good_mon-debug -c mon -- ceph-mon \
+    --fsid=$ceph_fsid \
+    --keyring=/etc/ceph/keyring-store/keyring \
+    --log-to-stderr=true \
+    --err-to-stderr=true \
+    --mon-cluster-log-to-stderr=true \
+    --log-stderr-prefix=debug \
+    --default-log-to-file=false \
+    --default-mon-cluster-log-to-file=false \
+    --mon-host=$ROOK_CEPH_MON_HOST \
+    --mon-initial-members=$ROOK_CEPH_MON_INITIAL_MEMBERS \
+    --id=$good_mon \
+    --foreground \
+    --public-addr=$good_mon_public_ip \
+    --setuser-match-path=/var/lib/ceph/mon/ceph-$good_mon/store.db \
+    --public-bind-addr=$ROOK_POD_IP \
+    --extract-monmap=$monmap_path
+
+  info_msg "Printing monmap"; \
+  KUBECTL_NS_CLUSTER exec deploy/rook-ceph-mon-$good_mon-debug -c mon -- monmaptool --print $monmap_path
+
+  # remove all the mons except the good one
+  for bad_mon in "${bad_mons[@]}"
+  do
+    info_msg "Removing mon $bad_mon"
+    KUBECTL_NS_CLUSTER exec deploy/rook-ceph-mon-$good_mon-debug -c mon -- monmaptool $monmap_path --rm $bad_mon
+  done
+
+  info_msg "Injecting the monmap"
+  KUBECTL_NS_CLUSTER exec deploy/rook-ceph-mon-$good_mon-debug -c mon -- ceph-mon \
+    --fsid=$ceph_fsid \
+    --keyring=/etc/ceph/keyring-store/keyring \
+    --log-to-stderr=true \
+    --err-to-stderr=true \
+    --mon-cluster-log-to-stderr=true \
+    --log-stderr-prefix=debug \
+    --default-log-to-file=false \
+    --default-mon-cluster-log-to-file=false \
+    --mon-host=$ROOK_CEPH_MON_HOST \
+    --mon-initial-members=$ROOK_CEPH_MON_INITIAL_MEMBERS \
+    --id=$good_mon \
+    --foreground \
+    --public-addr=$good_mon_public_ip \
+    --setuser-match-path=/var/lib/ceph/mon/ceph-$good_mon/store.db \
+    --public-bind-addr=$ROOK_POD_IP \
+    --inject-monmap=$monmap_path
+  info_msg "Finished updating the monmap!"
+  set -eu
+
+  info_msg "Printing final monmap"
+  KUBECTL_NS_CLUSTER exec deploy/rook-ceph-mon-$good_mon-debug -c mon -- monmaptool --print $monmap_path
+
+  info_msg "Restoring the mons in the rook-ceph-mon-endpoints configmap to the good mon"
+  KUBECTL_NS_CLUSTER patch configmaps rook-ceph-mon-endpoints --type json --patch "[{ op: replace, path: /data/data, value: $good_mon=$good_mon_public_ip:$good_mon_port }]"
+
+  info_msg "Stopping the debug pod for mon $good_mon"
+  run_stop_debug rook-ceph-mon-$good_mon
+
+  info_msg "Check that the restored mon is responding"
+  wait_for_mon_status_response
+
+  info_msg "Purging the bad mons: ${bad_mons[*]}"
+  for bad_mon in "${bad_mons[@]}"
+  do
+    info_msg "purging old mon: $bad_mon"
+    KUBECTL_NS_CLUSTER delete deploy rook-ceph-mon-$bad_mon
+    KUBECTL_NS_CLUSTER delete svc rook-ceph-mon-$bad_mon
+  done
+
+  info_msg "Mon quorum was successfully restored to mon $good_mon"
+
+  prompt_to_continue_or_cancel "Start up the operator and expand to full mon quorum again?" "yes"
```

Comment:
```suggestion
  prompt_to_continue_or_cancel "Start up the operator and expand to full mon quorum again?" "yes"
```
since Mon are successfully restored(as we are logging above) why we are asking for starting the operator up again, we should be doing every time once mons are up?

Reply:
It's a good question, it just felt like a good thing to pause and say "I did my job to restore quorum to a single mon, are you sure you're ready to start the operator up again?"

Reply:
if i understand correctly at this point we have a single monitor running right ? If the user cancels this prompt, they'll know that they're taking a risk to run with a single mon, maybe we should add some warning here if at a point they have a cluster running with a single monitor. 
And they would need to start and scale up the operator manually right in such situation.

Reply:
Yeah we don't want them to risk staying with a single mon. I think the "press any key to continue" will solve this.